st2client/tests/unit/test_keyvalue.py

```diff
@@ -0,0 +1,284 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import json
+import logging
+import mock
+import os
+import requests
+import tempfile
+import yaml
+
+from tests import base
+from st2client import shell
+
+LOG = logging.getLogger(__name__)
+
+KEYVALUE = {
+    'id': 'kv_name',
+    'name': 'kv_name.',
+    'value': 'super cool value',
+    'scope': 'system'
+}
+
+KEYVALUE_USER = {
+    'id': 'kv_name',
+    'name': 'kv_name.',
+    'value': 'super cool value',
+    'scope': 'system',
+    'user': 'stanley'
+}
+
+KEYVALUE_SECRET = {
+    'id': 'kv_name',
+    'name': 'kv_name.',
+    'value': 'super cool value',
+    'scope': 'system',
+    'secret': True
+}
+
+KEYVALUE_TTL = {
+    'id': 'kv_name',
+    'name': 'kv_name.',
+    'value': 'super cool value',
+    'scope': 'system',
+    'ttl': 100
+}
+
+KEYVALUE_ALL = {
+    'id': 'kv_name',
+    'name': 'kv_name.',
+    'value': 'super cool value',
+    'scope': 'system',
+    'user': 'stanley',
+    'secret': True,
+    'ttl': 100
+}
+
+KEYVALUE_MISSING_NAME = {
+    'id': 'kv_name',
+    'value': 'super cool value'
+}
+
+KEYVALUE_MISSING_VALUE = {
+    'id': 'kv_name',
+    'name': 'kv_name.'
+}
+
+
+class TestKeyValueBase(base.BaseCLITestCase):
```

Comment:
Perhaps you could also utilize `CleanFilesTestCase` class and `to_delete_files` attribute of that class.

You can grep code for some of the examples :)